src/framework/preferences/js/SelfVoicingEnactor.js

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2013-2015 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+var fluid_2_0 = fluid_2_0 || {};
+
+(function ($, fluid) {
+    "use strict";
+
+    /*******************************************************************************
+     * speak
+     *
+     * An enactor that is capable of speaking text.
+     * Typically this will be used as a base grade to an enactor that supplies
+     * the text to be spoken.
+     *******************************************************************************/
+
+    fluid.defaults("fluid.prefs.enactor.speakEnactor", {
+        gradeNames: ["fluid.prefs.enactor", "autoInit"],
+        preferenceMap: {
+            "fluid.prefs.speak": {
+                "model.value": "default"
+            }
+        },
+        model: {
+            value: false
+        },
+        events: {
+            onStart: null,
```

Comment:
If this were done via model relay from the model from the TTS component, we could eliminate all these listeners and event definitions etc. This of course would require cindyli's prefs framework update to go in, but I think that that was extremely close to being ready.